mir_eval/transcription.py

```diff
@@ -138,6 +138,173 @@ def validate(ref_intervals, ref_pitches, est_intervals, est_pitches):
                          "value")
 
 
+def validate_intervals(ref_intervals, est_intervals):
+    """Checks that the input annotations to a metric look like time intervals,
+    and throws helpful errors if not.
+
```

Comment:
The data may be well-formed, but I think it's worth issuing a warning when the set is empty, because it's a corner case that affects many evaluation metrics. Do you think it's better to leave it out of the validation function and move the warnings into each individual metric function? We're already issuing warnings about empty annotations in a few other validation function across `mir_eval`.